### PR TITLE
Donut chart improvements

### LINKED ISSUE #1465: Expose colour and tooltip properties for each segment in the DonutChart component
**Describe the problem/challenge you have**

In starboard-octant-plugin, which is written in Go, we'd like to use the DonutChart component to visualise a vulnerability summary. However, we cannot map predefined colours (red, orange, green) to all severity levels (CRITICAL,HIGH,MEDIUM,LOW,UNKNOWN)

octant-plugin

**Describe the solution you'd like**

Make the DonutChart component more generic so it can represent Pod or Node status, as well as such custom thing such as vulnerability summaries.

**Anything else you would like to add:**

N/A

**Environment:**

- Octant version (use `octant version`): 0.16.0
- Kubernetes version (use `kubectl version`): any
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): any

### PR DESCRIPTION
**Pull request by mklanjsek**
This change improves donut chart interactivity and adds following new functionality:

- Donut chart now supports custom colors on top of existing support for predefined types,
- Added thickness property for whole chart as well as individual segments,
- Improved hover indication and added tooltip for selected segment,
- Fixed chart sizing and some other small issues,

Signed-off-by: Milan Klanjsek <mklanjsek@pivotal.io>

**Which issue(s) this PR fixes**
- Fixes #1465

### CODE REVIEW

**Comment by mklanjsek** [comment]
Note that for single segment charts we show small gap at the end of the segment. This is due to limitation of svg arc functionality and I decided not to work around it because it's consistent with the way we show gaps between segments. Could be addressed in future if deemed necessary.

File: web/src/app/modules/shared/components/presentation/donut-chart/donut-chart.component.ts [hunk 1]
```diff
@@ -73,25 +75,132 @@ export class DonutChartComponent extends AbstractViewComponent<DonutChartView> {
...
+        this.tooltipX = `${event.offsetX}px`;
+        this.tooltipY = `${
+          event.offsetY -
+          event.target.ownerSVGElement.clientHeight -
+          scrollOffset -
+          48
+        }px`;
+        this.tooltipText = segment.description
+          ? segment.description
+          : `${segment.count} ${label} with status ${segment.status}`;
```
Comment (GuessWhoSamFoo) [inline web/src/app/modules/shared/components/presentation/donut-chart/donut-chart.component.ts:163]: Node status can still return as an empty string if not set

File: pkg/view/component/donut_chart.go [hunk 2]
```diff
@@ -63,6 +67,10 @@ func (dc *DonutChart) SetSize(size DonutChartSize) {
 	dc.Config.Size = size
 }
 
+func (dc *DonutChart) SetThickness(thickness int) {
+	dc.Config.Thickness = thickness
```
Comment (GuessWhoSamFoo) [inline pkg/view/component/donut_chart.go:71]: As a plugin developer, how will I know what is a reasonable number for thickness?

Negative thickness also makes interesting charts, but maybe it shouldn't be allowed
image

**Review by GuessWhoSamFoo — CHANGES_REQUESTED** [review]
The current model doesn't lend well to generalized use cases of donut charts (e.g. showing percentages). Not a big deal here, but something to think about as other graphs are implemented

File: web/src/app/modules/shared/components/presentation/donut-chart/donut-chart.component.ts (hunk 1 above)
Comment (mklanjsek) [reply web/src/app/modules/shared/components/presentation/donut-chart/donut-chart.component.ts]: Good point, made a change to omit the status if not available.

File: pkg/view/component/donut_chart.go (hunk 2 above)
Comment (mklanjsek) [reply pkg/view/component/donut_chart.go:73]: Right. Trimming this to [2-100] now. Also added comment to `SetThickness` method and story.

**Comment by mklanjsek** [comment]
@GuessWhoSamFoo PTAL

**Review by GuessWhoSamFoo — APPROVED** [review]

**Pull request merged**